### PR TITLE
i#5368: Replace disrespectful terms "master" and "slave"

### LINKED ISSUE #5368: Replace disrespectful terms "Master" and "Slave"
This is to identify and replace the term "Master" from the codebase/documentation where possible.

### PR DESCRIPTION
**Pull request by bete0**
Replaces "master" and "slave" with "main" (or "manager") and "worker"
respectively where possible. Also renames files containing these terms
with the corresponding inclusive terms.

Fixes #5368

### CODE REVIEW

File: core/unix/signal.c [hunk 1]
```diff
@@ -7760,7 +7760,7 @@ sig_detach_go_native(sig_detach_info_t *info)
     ASSERT_NOT_REACHED();
 }
 
-/* Sets this (slave) thread to detach by directly returning from the signal. */
+/* Sets this (worker) thread to detach by directly returning from the signal. */
```
Comment (derekbruening) [inline core/unix/signal.c:7763]: Hmm, to me "worker" implies a thread pool or something of threads dedicated to a task, while this is an app thread but is just not the one orchestrating the detach. Maybe "secondary" or "follower" instead?

File: ext/drmgr/CMakeLists.txt [hunk 2]
```diff
@@ -70,7 +70,7 @@ configure_drmgr_target(drmgr)
 
 # Must be shared to ensure only one copy, since can be used by
 # the client and by multiple other libraries, some of which may
-# be shared: the whole point is to have one master drmgr.
+# be shared: the whole point is to have one main drmgr.
```
Comment (derekbruening) [inline ext/drmgr/CMakeLists.txt:73]: How about "unique"

File: ext/drreg/drreg.c [hunk 3]
```diff
@@ -2478,7 +2478,7 @@ drreg_init(drreg_options_t *ops_in)
         return DRREG_ERROR_INVALID_PARAMETER;
 
     /* Instead of allowing only one drreg_init() and all other components to be
-     * passed in scratch regs by a master, which is not always an easy-to-use
+     * passed in scratch regs by a main, which is not always an easy-to-use
```
Comment (derekbruening) [inline ext/drreg/drreg.c:2481]: Hmm to me it sounds off -- the original seems off too.  I would do `s/by a main//`

File: core/synch.c [hunk 4]
```diff
@@ -2248,7 +2248,7 @@ detach_on_permanent_stack(bool internal, bool do_cleanup, dr_stats_t *drstats)
         dynamo_thread_exit_pre_client(my_dcontext, my_tr->id);
     }
 
-    LOG(GLOBAL, LOG_ALL, 1, "Detach: Letting slave threads go native\n");
+    LOG(GLOBAL, LOG_ALL, 1, "Detach: Letting worker threads go native\n");
```
Comment (derekbruening) [inline core/synch.c:2251]: Like below, I would replace this with whatever is chosen for detach there: "secondary" or "follower"

**Review by derekbruening — APPROVED** [review]
Just a few suggestions for different replacements in some cases.

File: core/unix/signal.c (hunk 1 above)
Comment (bete0) [reply core/unix/signal.c]: Done; changed to "secondary".

File: ext/drmgr/CMakeLists.txt (hunk 2 above)
Comment (bete0) [reply ext/drmgr/CMakeLists.txt]: Done.

File: ext/drreg/drreg.c (hunk 3 above)
Comment (bete0) [reply ext/drreg/drreg.c]: Done.

File: core/synch.c (hunk 4 above)
Comment (bete0) [reply core/synch.c]: Done; changed to "secondary"

**Comment by johnfxgalea** [comment]
We should also change the DR's default branch to main, modifying CI scripts accordingly.

**Comment by derekbruening** [comment]
> > We should also change DR's default branch to main, modifying CI scripts accordingly (maybe separate PR?).
> 
> Agreed. We have had a discussion to do that in a way that won't negatively affect contributors and users.

I believe this complicates outstanding pull requests and other users and would have to be done carefully and probably with announcements.  I assumed we would at least first do a cleanup pass through the open pull requests (needs a cleanup pass in any case).

**Pull request merged**